### DISCUSSION TITLE
[V3] Commons validation messages

### DISCUSSION THREAD

**Discussion by m15studio** [livewire/livewire#6140] [1. Help] [answered]
Hello,
I'm trying to define common messages in form validation rules.

eg. I would a custom message on "required"
`"required" => ":attribute is required in this application. this is a custom message"`

but I wouldn't to declare same message for each #[Rule]

If I declare message as usually 

`protected $messages = [
"required" => ":attribute is required in this application. this is a custom message"
]`

when I submit form the validation function $this->validate() ignores completely custom message

**Comment by m15studio** [comment]
I found a solution but I am not sure that it is a correct way..

I've moved rules(), messages() and validationAttributes() from Form Object to Component Object.
In this way validation works with custom rules and messages.

Anyone else in this situation?

**Reply by viliusvsx** [reply, marked as answer]
This issue is discussed in this PR [#5904](https://github.com/livewire/livewire/pull/5904). Hope this get merged soon
